pkg/config/template.go

```diff
@@ -85,6 +88,22 @@ func NewTemplateCMManager(cmclient corev1.ConfigMapInterface, prNumber int, logg
 	}
 }
 
+func (c *TemplateCMManager) createCM(cm *v1.ConfigMap) error {
+	l := &cm.ObjectMeta.Labels
+	if *l == nil {
+		*l = map[string]string{}
+	}
+	(*l)[createByRehearse] = "true"
+	(*l)[rehearseLabelPull] = strconv.Itoa(c.prNumber)
+	if _, err := c.cmclient.Create(cm); err == nil || !kerrors.IsAlreadyExists(err) {
+		return err
+	}
+	if _, err := c.cmclient.Update(cm); err != nil {
```

Comment:
We want to `Get` before we `Update` and make sure to do retries to ensure we don't get into the odd version issues

Reply:
I merely extracted this to a separate function without changing the behavior (i.e. this is what we currently do in production), but OK.

Reply:
We can handle it in a follow-up, then :)

Reply:
I think we can remove the `Update`. Since we are hasing the name of the cm.

Reply:
Also in a follow-up so this can be just a move